runtime/gc_modron_startup/mminit.cpp

```diff
@@ -3044,12 +3044,22 @@ j9gc_reinitializeDefaults(J9VMThread* vmThread)
 	J9JavaVM* vm = vmThread->javaVM;
 	bool result = true;
 
+	PORT_ACCESS_FROM_JAVAVM(vm);
+
 	extensions->gcThreadCountForced = false;
+	extensions->parSweepChunkSize = 0;
 
 	if (!gcParseReconfigurableArguments(vm, vm->checkpointState.restoreArgsList)) {
 		result = false;
 	}
 
+	/* Verify the GC thread count based on current restore limitation - threads don't shutdown at restore. */
+	if (extensions->gcThreadCountForced && (extensions->gcThreadCount < extensions->dispatcher->threadCountMaximum())) {
```

Comment:
Hm, we should idally allow this. Is this a conscious dicision of we just had no time to handle his scenario? What will happen if we proceed (and don't shutdown the excess threasds)?

Reply:
This was a conscious decision, made when we didn't have options at restore time. Give the original problem addressed by this work, It didn't make sense to shutdown threads at restore time.

Reply:
>What will happen if we proceed (and don't shutdown the excess threads)?

The Expand thread pool and Contract thread pool are 2 separate APIs, called independently at depending on checkpoint or restore phase.

Reply:
The dispatcher ensures the GC thread count being expanded to is greater than the checkpoint thread count.

Reply:
Ok, let's go with this limitation for now, but I think there is value (limited, but still) allowing contraction on restore side. For  completeness we could even allow expansion on checkpoint side (probably even less value). In that case,  there could be just one public API that should both contraction and expansion.